.github/workflows/aws-ci.yaml

```diff
@@ -0,0 +1,107 @@
+name: CI tests AWS
+on:
+  pull_request_target:
+    types: [labeled, synchronize]  # Triggered when the PR is labeled or synchronized
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+jobs:
+  deploy-ec2:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Remove ok-to-test label if new commit
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const label = "ok-to-test";
+            const pr = context.payload.pull_request;
+            const labels = pr.labels.map(l => l.name);
+            if (labels.includes(label)) {
+              await github.rest.issues.removeLabel({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: pr.number,
+                name: label
+              });
+              console.log(`Removed ${label} due to new push`);
+            }
+      - name: Checkout the code
+        uses: actions/checkout@v3
+        with:
+          repository: k8snetworkplumbingwg/ptp-operator
+          ref: main
+          path: ptp-operator
+      
+      - name: checkout the Current SHA for the linuxptp-daemon
+        id: sha
+        run: |
+          sed -i '/^RUN git clone -b main https:\/\/github.com\/k8snetworkplumbingwg\/linuxptp-daemon.git  \/go\/src\/github.com\/k8snetworkplumbingwg\/linuxptp-daemon$/c\
+          RUN git clone -b main https://github.com/k8snetworkplumbingwg/linuxptp-daemon.git  /go/src/github.com/k8snetworkplumbingwg/linuxptp-daemon\nRUN git checkout ${{ github.sha }}' ./ptp-operator/ptp-tools/Dockerfile.lptpd
+          cat ./ptp-operator/ptp-tools/Dockerfile.lptpd
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+            role-to-assume: arn:aws:iam::058264082309:role/github-oidc-setup-david-Role-Kk44xkca6vIm
+            role-session-name: ${{ github.run_id }}
+            aws-region: us-east-1
+
+      - name: Launch EC2 instance
+        id: launch
+        run: |
+          INSTANCE_ID=$(aws ec2 run-instances \
+            --image-id ami-0f7270e9a0e492eac \
+            --subnet-id subnet-04ee218fe8f32cad5 \
+            --instance-type m6a.xlarge \
+            --security-group-ids sg-05e700edc2ed0a2d5 \
+            --tag-specifications 'ResourceType=instance,Tags=[{Key=Name,Value=ptp-operator PR-${{ github.event.number }}}]' \
+            --query 'Instances[0].InstanceId' \
+            --key-name ptp-sshkey-1 \
+            --output text)
+
+          echo "instance_id=$INSTANCE_ID" >> $GITHUB_ENV
```

Comment:
you are saving instance_id in to github env and then you never use it 
why don't you use ${{ env.instance_id }} instead?
Don't you need to re export?
instance_id=${{ env.instance_id }}

Reply:
I do use instance_id later on when deleting the instance see:
```
      - name: Cleanup - Delete EC2 instance
        if: always()  # Ensure cleanup runs even if previous steps fail
        run: |
            echo "Cleaning up EC2 INSTANCE_ID=$instance_id"
            aws ec2 terminate-instances --instance-ids $instance_id
```